plugins/provisioners/ansible/provisioner/guest.rb

```diff
@@ -128,6 +128,8 @@ def generate_inventory_machines
               else
                 machines += "#{machine_name}\n"
               end
+              host_vars = get_inventory_host_vars_string(machine_name)
+              machines.sub!(/\n$/, " #{host_vars}\n") if host_vars
```

Comment:
How important is the `$`! I first think it was a bug, and cannot believe that my test was successful ;-)

In the future, I think that it might be good to append the `host_vars` values in a the `base.rb` shared parts. But to achieve that, `generate_inventory_machines` should then return an array of hosts instead of the "rendered text".

(This comment is here for the records. No change required in this review)